src/components/form-elements/draft-js-mention-selector/DraftJSMentionSelector.js

```diff
@@ -87,15 +87,47 @@ class DraftJSMentionSelector extends React.Component<Props, State> {
      * @returns {void}
      */
     componentWillReceiveProps(nextProps: Props) {
-        const { contacts } = nextProps;
+        const { editorState } = this.props;
+        const { contacts, editorState: nextEditorState } = nextProps;
+        let nextState = {};
 
         if (contacts) {
-            this.setState({ contacts }, () => {
+            nextState = { ...nextState, contacts };
+        }
+
+        // Only check if operating in the mode where EditorState is received as a
+        // prop vs internalEditorState stored in state
+        if (editorState) {
+            // Detect case where controlled EditorState is created anew and empty.
+            // If next editorState is empty and the current editorState is not empty
+            // that means it is a new empty state and this component should not be marked dirty
+            if (this.isEditorStateEmpty(nextEditorState) && !this.isEditorStateEmpty(editorState)) {
```

Comment:
Maybe make these constants, might be a wee bit more readable.

isCurrentEditorStateEmpty
isNextEditorStateEmpty